test/TensorFlowNET.Graph.UnitTest/ControlFlowTest/WhileContextTestCase.cs

```diff
@@ -29,7 +30,7 @@ private void _testWhileContextHelper(int maximum_iterations)
             var b = new Func<Tensor, Tensor>(x => math_ops.add(x, 1, name: "c"));
             //control_flow_ops.while_loop(
             //      c, b, i , maximum_iterations: tf.constant(maximum_iterations));
-            foreach (Operation op in sess.graph.get_operations())
+            foreach (Operation op in sess.Single().graph.get_operations())
```

Comment:
@Oceania2018 I'm not sure how exactly IEnumerable should work there since Python generator produces only one instance. Anyway this test wasn't working and it is still not implemented yet.

Reply:
The test didn't come through.
```shell
   Test method TensorFlowNET.UnitTest.Gradient.GradientTest.testBoundaryContinue threw exception: 
Tensorflow.ValueError: Cannot use the default session to evaluate tensor: the tensor's graph is different from the session's graph. Pass an explicit session to `eval(session=sess)`.
```

Reply:
Sorry, it was flaky. Now it should work.

Reply:
> @Oceania2018 I'm not sure how exactly IEnumerable should work there since Python generator produces only one instance. Anyway this test wasn't working and it is still not implemented yet.
Since the different behavior of `yield` in Python and C#, If cached_session only need to have one session, I think it is better not to use `yield`? And then you can use `using` to replace `with` in GradientTest.cs

Reply:
> @Oceania2018 I'm not sure how exactly IEnumerable should work there since Python generator produces only one instance. Anyway this test wasn't working and it is still not implemented yet.
Since the different behavior of `yield` in Python and C#, If cached_session only need to have one session, I think it is better not to use `yield`? And then you can use `using` to replace `with` in GradientTest.cs

Reply:
Could, please, clarify why `yield` used in Python in this case?
Anyway fixed.